### PR TITLE
add mrfioc2 and devlib2 support

### PR DESCRIPTION
**Pull request by agaget**
Using of the standard driver, customs modifications from CEA will be included at one point one day. It is not necessary for standard using.

In draft, because it needs to be tested with hardware.

### CODE REVIEW

**Comment by agaget** [comment]
Not a draft, anymore. 2.6.0 (with my mask features) fully tested in laboratory

**Comment by agaget** [comment]
can it be reviewed ?

**Comment by agaget** [comment]
> can it be reviewed please ?

or no ?

**Review by stephane-cea — COMMENTED** [review]
It looks good to me!

But I think I'm not the main reviewer here, I'll let @minijackson approve and merge.

File: pkgs/epnix/support/devlib2/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "epics-modules";
+    repo = "devlib2";
+    rev = version;
+    sha256 = "sha256-5rjilz+FO6ZM+Hn7AVwyFG2WWBoBUQA4WW5OHhhdXw4=";
+  };
+
+  meta = {
+    description = "devLib2 - Library for direct MMIO access to PCI and VME64x";
+    homepage = "https://github.com/epics-modules/devlib2";
+    license = lib.licenses.lgpl21Plus;
```
Comment (minijackson) [inline pkgs/epnix/support/devlib2/default.nix:28]: devlib2 seems to be developed under the EPICS license: https://github.com/epics-modules/devlib2/blob/master/LICENSE

File: pkgs/epnix/support/devlib2/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  fetchFromGitHub,
+  fetchpatch,
+  local_config_site ? {},
+  local_release ? {},
+}:
+mkEpicsPackage rec {
+  pname = "devlib2";
+  version = "2.12";
+  varname = "DEVLIB2";
+  doCheck = false;
```
Comment (minijackson) [inline pkgs/epnix/support/devlib2/default.nix:14]: If tests don't pass, can you add a comment explaining what happens?

File: pkgs/epnix/support/mrfioc2/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  fetchFromGitHub,
+  fetchpatch,
+  local_config_site ? {},
+  local_release ? {},
+}:
+mkEpicsPackage rec {
+  pname = "mrfioc2";
+  version = "2.6.0";
+  varname = "MRFIOC2";
+  doCheck = false;
```
Comment (minijackson) [inline pkgs/epnix/support/mrfioc2/default.nix:15]: If tests don't pass, can you add a comment explaining what happens?

File: pkgs/epnix/support/mrfioc2/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  propagatedBuildInputs = with epnix.support; [devlib2];
+
+  meta = {
+    description = "Driver EPICS for MRF cards";
+    homepage = "https://github.com/epics-modules/mrfioc2";
+    license = epnixLib.licenses.epics;
+    maintainers = with epnixLib.maintainers; [agaget];
+  };
```
Comment (minijackson) [inline pkgs/epnix/support/mrfioc2/default.nix:33]: `meta` should be the last attribute

File: pkgs/epnix/support/mrfioc2/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  postInstall = ''
+    if [[ -d iocBoot ]]; then
+      cp -rafv iocBoot -t "$out"
+    fi
+
+  '';
```
Comment (minijackson) [inline pkgs/epnix/support/mrfioc2/default.nix:40]:
```suggestion
  postInstall = ''
    if [[ -d iocBoot ]]; then
      cp -rafv iocBoot -t "$out"
    fi
  '';
```

File: lib/maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -62,4 +62,8 @@ See `<nixpkgs/maintainers/scripts/check-maintainer-github-handles.sh>` for an ex
     email = "stephane.tzvetkov@cea.fr";
     name = "Stéphane Tzvetkov";
   };
+  agaget = {
```
Comment (minijackson) [inline lib/maintainers/maintainer-list.nix:65]: Can you put yourself at the top? The comment above says to keep the list alphabetically sorted

**Review by minijackson — COMMENTED** [review]
Thanks! Only minor comments. Could you split these changes into 2 commits? One for devlib2 / adding yourself as maintainer, and the other one for mrfioc2?

File: lib/maintainers/maintainer-list.nix (hunk 6 above)
Comment (agaget) [reply lib/maintainers/maintainer-list.nix:65]: I'm always at the top 👓

**Comment by agaget** [comment]
I think, I have resolved everything

**Review by minijackson — APPROVED** [review]
Thanks!

**Pull request merged**